src/page-modules/assistant/trip/index.tsx

```diff
@@ -96,68 +105,87 @@ export default function Trip({ tripQuery, trip, nonTransitTrips }: TripProps) {
 
       <Button
         className={style.fetchButton}
-        onClick={() =>
-          fetchNextTripPatterns(
-            tripQuery.from!,
-            tripQuery.to!,
-            tripQuery.transportModeFilter || undefined,
-          )
-        }
+        onClick={() => fetchNextTripPatterns()}
         title={t(PageText.Assistant.trip.fetchMore)}
-        state={isFetchingTripPatterns ? 'loading' : undefined}
+        state={isFetchingNextTripPatterns ? 'loading' : undefined}
       />
     </>
   );
 }
 
-function useTripPatterns(initialTrip: TripData, searchTime: SearchTime) {
-  const [tripPatterns, setTripPatterns] = useState(
-    tripPatternsWithTransitionDelay(initialTrip.tripPatterns),
-  );
-  const [cursor, setCursor] = useState(
-    getCursorBySearchMode(initialTrip, searchTime.mode),
-  );
+function useTripPatterns(query: FromToTripQuery) {
+  const [tripPatterns, setTripPatterns] = useState<
+    (TripPatternType & { transitionDelay: number })[]
+  >([]);
+  const [nonTransitTrips, setNonTransitTrips] = useState<
+    NonTransitTripData | undefined
+  >(undefined);
+  const [cursor, setCursor] = useState<string>();
   const [isFetchingTripPatterns, setIsFetchingTripPatterns] = useState(false);
+  const [isFetchingNextTripPatterns, setIsFetchingNextTripPatterns] =
+    useState(false);
+
+  useEffect(() => {
+    if (!query.from || !query.to) return;
+    const fetch = async () => {
+      setIsFetchingTripPatterns(true);
+      const tripQuery = createTripQuery(
+        query,
+        getInitialTransportModeFilter(null),
+      );
+      const trip = await getTripPatterns(tripQuery);
+      const newTripPatternsWithTransitionDelay =
+        tripPatternsWithTransitionDelay(trip.tripPatterns);
+
+      setTripPatterns(newTripPatternsWithTransitionDelay);
+      setCursor(getCursorBySearchMode(trip, query.searchTime.mode));
+      setIsFetchingTripPatterns(false);
+    };
+    fetch();
+  }, [query]);
```

Comment:
Maybe I should have thought of this earlier, but thinking about this. Maybe useSWR would be a better approach here than a clean fetch call? swr has Fallback-components which can support SSR. So if Fallback component has data from SSR it will be populated from server, if not it will do client cache. This way we can inject cached results from server when navigating back if that makes sense.